statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,441 @@
+
+
+def _pickle_method(m):
+    """Handle pickling issues with class instance."""
+    if m.im_self is None:
+        return getattr, (m.im_class, m.im_func.func_name)
+    else:
+        return getattr, (m.im_self, m.im_func.func_name)
+
+
+copyreg.pickle(types.MethodType, _pickle_method)
+
+
+class HdrResults(object):
+
+    """Wrap results and pretty print them."""
+
+    def __init__(self, kwds):
+        self.__dict__.update(kwds)
+
+    def __repr__(self):
+        msg = ("HDR boxplot summary:\n"
+               "-> median:\n{}\n"
+               "-> 50% HDR (max, min):\n{}\n"
+               "-> 90% HDR (max, min):\n{}\n"
+               "-> Extra quantiles (max, min):\n{}\n"
+               "-> Outliers:\n{}"
+               ).format(self.median, self.hdr_50, self.hdr_90,
+                        self.extra_quantiles, self.outliers)
+
+        return msg
+
+
+def _inverse_transform(pca, data):
+    """Inverse transform on PCA.
+
+    Use PCA's `project` method by temporary replacing its factors with
+    `data`.
+
+    Parameters
+    ----------
+    pca : statsmodels Principal Component Analysis instance
+        The PCA object to use.
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+
+    Returns
+    -------
+    projection : array
+        nobs by nvar array of the projection onto ncomp factors
+
+    """
+    factors = pca.factors
+    pca.factors = data.reshape(-1, factors.shape[1])
+    projection = pca.project()
+    pca.factors = factors
+    return projection
+
+
+def _curve_constrain(x, idx, sign, band, pca, ks_gaussian):
+    """Find out if the curve is within the band.
+
+    The curve value at :attr:`idx` for a given PDF is only returned if
+    within bounds defined by the band. Otherwise, 1E6 is returned.
+
+    Parameters
+    ----------
+    x : float
+        Curve in reduced space.
+    idx : int
+        Index value of the components to compute.
+    sign : int
+        Return positive or negative value.
+    band : list of float
+        PDF values `[min_pdf, max_pdf]` to be within.
+    pca : statsmodels Principal Component Analysis instance
+        The PCA object to use.
+    ks_gaussian : KDEMultivariate instance
+
+    Returns
+    -------
+    value : float
+        Curve value at `idx`.
+
+    """
+    x = x.reshape(1, -1)
+    pdf = ks_gaussian.pdf(x)
+    if band[0] < pdf < band[1]:
+        value = sign * _inverse_transform(pca, x)[0][idx]
+    else:
+        value = 1E6
+    return value
+
+
+def _min_max_band(args):
+    """Min an max values at `idx`.
+
+    Global optimization to find the extrema per component.
+
+    Parameters
+    ----------
+    args: list
```

Comment:
Why do you use args and not explicit arguments?
Because of the use in pool.map, I guess, checking the usage

Reply:
Indeed, this is due to `pool.map` which only accepts a single argument.